examples/guides/services/healthcare-service/tests/clemency-hospital-service-test.bal

```diff
@@ -0,0 +1,442 @@
+// Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+import ballerina/http;
+import ballerina/io;
+import ballerina/log;
+
+http:Client clientEPclemency = new("http://localhost:9090/clemency/categories");
+
+# Description: This test verifies if an appointment can be reserved successfully. 
+# + dataset - dataset Parameter Description
+@test:Config {
+    dataProvider: "testReserveAppointmentDataProvider",
+    dependsOn: ["testAddDoctor"]
+}
+function testReserveAppointment(json dataset, json expectedStrings) {
+    // set the json payload
+    http:Request request = new;
+    request.setPayload(dataset);
+
+    // sending the post request to the endpoint
+    http:Response | error response = clientEPclemency->post("/surgery/reserve", request);
+
+    if (response is http:Response) {
+        string doctor = dataset.doctor.toString();
+        var responseStatusCode = response.statusCode;
+        var expectedStatusCode = expectedStrings.statusCode;
+        string expectedResponseText = expectedStrings.responseMessage.toString();
+
+        if (doctor == "T Uyanage") {
```

Comment:
Why do we test the incoming input dataset here?

Reply:
Updated the negative test cases.